internal/nix/run.go

```diff
@@ -24,9 +25,13 @@ func RunScript(projectDir, cmdWithArgs string, env map[string]string) error {
 		envPairs = append(envPairs, fmt.Sprintf("%s=%s", k, v))
 	}
 
+	// Wrap in quotations since the command's path may contain spaces.
+	cmdWithArgs[0] = "\"" + cmdWithArgs[0] + "\""
+	cmdWithArgsStr := strings.Join(cmdWithArgs, " ")
```

Comment:
You can use `fmt.Springf("%q", s)` to surround a string in quotes (in a way that will properly escape stuff in the string).

godocs:
`%q	a double-quoted string safely escaped with Go syntax`

thought, It makes me a tiny bit nervous, what if the arg is already quoted?

I do wonder if this is the wrong place to do this. Specifically, `shellgen.ScriptPath` gets called in a bunch of places then possibly used without quotes. For example, `EnvExports` will return a broken (non quoted) source line if `RunHooks` is true. (This is why you had to surround `HooksFilePath` in quotes in shellrc here https://github.com/jetpack-io/devbox/pull/1920/files )

We could rename `shellgen.ScriptPath` to `shellgen.QuotedScriptPath` instead and just make it clear it's quoted? Not sure if this is best.